src/mgr/DaemonServer.h

```diff
@@ -111,6 +112,22 @@ class DaemonServer : public Dispatcher, public md_config_obs_t
   void tick();
   void schedule_tick_locked(double delay_sec);
 
+  class OSDPerfMetricCollectorListener :
+      public OSDPerfMetricCollector::Listener {
+  public:
+    OSDPerfMetricCollectorListener(DaemonServer *server)
+      : server(server) {
+    }
+    void handle_query_updated() override {
+      server->handle_osd_perf_metric_query_updated();
+    }
+  private:
+    DaemonServer *server;
+  };
+  OSDPerfMetricCollectorListener osd_perf_metric_collector_listener;
```

Comment:
Nit: I'm on the fence since the existing code doesn't follow the Ceph coding style, but I feel like new member variables should be prefixed w/ ```m_```